src/main/java/teammates/ui/webapi/action/GetInstructorPrivilegeAction.java

```diff
@@ -26,49 +91,74 @@ public void checkSpecificAccessControl() {
+
+        if (instructorRole != null) {
+            // fetching privilege for a particular instructor role.
+            InstructorPrivilegeData rolePrivilege = INSTRUCTOR_PRIVILEGES.getOrDefault(instructorRole, null);
+            if (rolePrivilege == null) {
+                return new JsonResult("Invalid instructor role.", HttpStatus.SC_BAD_REQUEST);
+            }
+            return new JsonResult(rolePrivilege);
+        }
+
         String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+        String sectionName = getRequestParamValue(Const.ParamsNames.SECTION_NAME);
         String feedbackSessionName = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME);
+
         InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.getId());
+        InstructorPrivilegeData response = new InstructorPrivilegeData();
 
-        InstructorPrivilegeResponse response = new InstructorPrivilegeResponse();
+        // course level privileges.
         response.setCanModifyCourse(
                 instructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_COURSE));
         response.setCanModifySession(
                 instructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION));
         response.setCanModifyStudent(
                 instructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT));
-        response.setCanSubmitSessionInSections(
-                instructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_SUBMIT_SESSION_IN_SECTIONS)
-                        || instructor.isAllowedForPrivilegeAnySection(
-                                feedbackSessionName, Const.ParamsNames.INSTRUCTOR_PERMISSION_SUBMIT_SESSION_IN_SECTIONS));
+        response.setCanModifyInstructor(
+                instructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR));
 
-        return new JsonResult(response);
-    }
+        if (sectionName == null) {
+            response.setCanViewStudentInSections(
+                    instructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS));
 
-    /**
-     * The output format of {@link GetInstructorPrivilegeAction}.
-     */
-    public static class InstructorPrivilegeResponse extends ApiOutput {
-        private boolean canModifyCourse;
-        private boolean canModifySession;
-        private boolean canModifyStudent;
-        private boolean canSubmitSessionInSections;
-
-        public void setCanModifyCourse(boolean canModifyCourse) {
-            this.canModifyCourse = canModifyCourse;
-        }
+            response.setCanSubmitSessionInSections(
+                    instructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_SUBMIT_SESSION_IN_SECTIONS)
+                            || instructor.isAllowedForPrivilegeAnySection(
```

Comment:
Make sure you have this `or` tested.

That is when `nstructor.isAllowedForPrivilege(...)` is false while `instructor.isAllowedForPrivilegeAnySection(...)` is true.

Reply:
Yes, I have designed the test data carefully to ensure this is covered.